exp/controllers/azuremangedcontrolplane_reconciler.go

```diff
@@ -140,6 +148,31 @@ func (r *azureManagedControlPlaneReconciler) Delete(ctx context.Context, scope *
 	return nil
 }
 
+func (r *azureManagedControlPlaneReconciler) reconcileNetwork(ctx context.Context, scope *scope.ManagedControlPlaneScope, managedClusterSpec *managedclusters.Spec) error {
+	vnets, err := r.vnetSvc.List(ctx, scope.ControlPlane.ManagedResourceGroup())
+	if err != nil {
+		return errors.Wrapf(err, "failed to list vnets")
+	}
+
+	if len(vnets) < 1 {
+		return fmt.Errorf("failed to find vnet in managed resource group '%s' for aks cluster", scope.ControlPlane.ManagedResourceGroup())
+	}
+
+	if vnets[0].Name == nil {
+		return errors.New("expected vnet returned by Azure to have a name, but was nil")
+	}
+
+	old := scope.ControlPlane.DeepCopyObject()
+
+	scope.ControlPlane.Status.VirtualNetwork = *vnets[0].Name
```

Comment:
what happens if there is more than 1? the assumption might work right now but it could become fragile if changes are made in the future to avoid vnet peerings and things like that, or simply if the RG is used by multiple clusters.

Reply:
we can probably fetch it out of the MC object, you are right

Reply:
oh right, the MC object doesn’t have it, but the agent pools have vnetSubnetID. Let’s sync up offline and I can also ask RP team if there’s any historical context on this one.